src/main/java/io/lettuce/core/json/DelegateJsonValue.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2024, Redis Ltd. and Contributors
+ * All rights reserved.
+ *
+ * Licensed under the MIT License.
+ *
+ * This file contains contributions from third-party contributors
+ * licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lettuce.core.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.lettuce.core.codec.RedisCodec;
+import io.lettuce.core.codec.StringCodec;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Implementation of the {@link JsonValue} that delegates most of it's dunctionality to the Jackson {@link JsonNode}.
+ *
+ * @param <K> Key type.
+ * @param <V> Value type.
+ * @author Tihomir Mateev
+ */
+class DelegateJsonValue<K, V> implements JsonValue<K, V> {
+
+    protected final RedisCodec<K, V> codec;
+
+    protected JsonNode node;
+
+    DelegateJsonValue(JsonNode node, RedisCodec<K, V> codec) {
+        this.codec = codec;
+        this.node = node;
+    }
+
+    @Override
+    public V toValue() {
+        byte[] result = node.toString().getBytes();
+        return codec.decodeValue(ByteBuffer.wrap(result));
```

Comment:
See my comment in `JsonValueListOutput`. I think we can "detach" codec and JSON to make all JSON interfaces/types concrete types and reduce the complexity of the code.

As I mentioned in another comment, it doesn't make sense to decode anything that was parsed by the JSON parser with the Lettuce value codec.

Reply:
As mentioned before, two things would result if your suggestion is applied:
- encoding / decoding of any (including large) JSON documents will be done inside the event loop, potentially slowing down processing in a large application with multiple threads working with the event loop.
- we would have to know how to transform the result from any codec to String. The Jackson (and all other JSON libraries I've seen) work with either byte array or String, but the Codec could decode data to any type the user found useful to them.